### PR TITLE
add .strict method which only returns GA spec'd utm params

### PR DESCRIPTION
**Pull request by sperand-io**
`utm.strict` only returns utm params that are spec'd by GA. 

(these ones:)

image

### CODE REVIEW

**Comment by sperand-io** [comment]
FYI @wcjohnson11 =D

@ndhoule plz review.

File: test/index.test.js [hunk 1]
```diff
@@ -42,3 +44,16 @@ describe('utm-params', function() {
     assert.deepEqual(params, { name: 'foo', source: 'baz' });
   });
 });
+
+describe('utm-params.strict', function() {
+    var params = utm.strict('?utm_source=source&utm_medium=medium&utm_term=term&utm_content=content&utm_campaign=name&utm_test=test&utm_fake=fake');
```
Comment (ndhoule) [inline test/index.test.js]: should be inside the `it` block

File: component.json [hunk 2]
```diff
@@ -8,7 +8,9 @@
     "index.js"
   ],
   "dependencies": {
-    "component/querystring": "*"
+    "component/querystring": "*",
```
Comment (ndhoule) [inline component.json]: Make this a non-wildcard dependency

File: index.js [hunk 3]
```diff
@@ -45,8 +47,26 @@ function utm(query){
...
+ * Get strict utm params - from the given `querystring`
+ *
+ * @param {String} query
+ * @return {Object}
+ * @api private
+ */
+
+function strict(query) {
+  return foldl(function(acc, val, key) {
+    if (includes(key, allowedKeys)) acc[key] = val;
```
Comment (ndhoule) [inline index.js]: This is a worst-case `n^2` operation, use a hash table to make this more efficient:

```js
var allowedKeys = {
  name: true,
  term: true,
  source: true,
  medium: true,
  content: true
};

//...
    if (allowedKeys.hasOwnProperty(key)) acc[key] = val;
```

**Comment by sperand-io** [comment]
@ndhoule thanks! good catch on that exponential operation. all comments accounted for and rebased.

**Pull request merged**